gobblin-aws/src/main/java/gobblin/aws/GobblinAWSClusterLauncher.java

```diff
@@ -0,0 +1,666 @@
+
+  public static final String CLUSTER_NAME_ASG_TAG = "ClusterName";
+  public static final String CLUSTER_ID_ASG_TAG = "ClusterId";
+  public static final String ASG_TYPE_ASG_TAG = "AsgType";
+  public static final String ASG_TYPE_MASTER = "master";
+  public static final String ASG_TYPE_WORKERS = "workers";
+
+  public static final String MASTER_ASG_NAME_PREFIX = "GobblinMasterASG_";
+  public static final String MASTER_LAUNCH_CONFIG_NAME_PREFIX = "GobblinMasterLaunchConfig_";
+  public static final String WORKERS_ASG_NAME_PREFIX = "GobblinWorkerASG_";
+  public static final String WORKERS_LAUNCH_CONFIG_PREFIX = "GobblinWorkerLaunchConfig_";
+
+  private final Config config;
+
+  private final HelixManager helixManager;
+  private final EventBus eventBus = new EventBus(GobblinAWSClusterLauncher.class.getSimpleName());
+  private volatile Optional<ServiceManager> serviceManager = Optional.absent();
+  private AWSClusterSecurityManager awsClusterSecurityManager;
+  private AWSSdkClient awsSdkClient;
+
+  private final Closer closer = Closer.create();
+
+  // AWS cluster meta
+  private final String clusterName;
+  private volatile Optional<String> clusterId = Optional.absent();
+
+  private volatile boolean stopped = false;
+  private final boolean emailNotificationOnShutdown;
+
+  // AWS Gobblin cluster common config
+  private final String awsRegion;
+  private final String awsConfDir;
+
+  // AWS Gobblin Master Instance config
+  private final String masterAmiId;
+  private final String masterInstanceType;
+  private final String masterJvmMemory;
+
+  // AWS Gobblin Worker Instance config
+  private final String workerAmiId;
+  private final String workerInstanceType;
+  private final String workerJvmMemory;
+  private final Integer minWorkers;
+  private final Integer maxWorkers;
+  private final Integer desiredWorkers;
+
+  private final Optional<String> masterJvmArgs;
+  private final Optional<String> workerJvmArgs;
+
+  private String masterPublicIp;
+
+  private final String nfsParentDir;
+  private final String masterJarsDir;
+  private final String masterConfLocalDir;
+  private final String masterS3ConfUri;
+  private final String masterS3ConfFiles;
+  private final String masterS3JarsUri;
+  private final String masterS3JarsFiles;
+
+  private final String workerJarsDir;
+  private final String workerConfLocalDir;
+  private final String workerS3ConfUri;
+  private final String workerS3ConfFiles;
+  private final String workerS3JarsUri;
+  private final String workerS3JarsFiles;
+
+  private final String libJarsDir;
+  private final String sinkLogRootDir;
+  private final String appWorkDir;
+
+  private String masterLaunchConfigName;
+  private String masterAutoScalingGroupName;
+
+  private String workerLaunchConfigName;
+  private String workerAutoScalingGroupName;
+
+  private final Optional<String> gobblinVersion;
+
+
+  public GobblinAWSClusterLauncher(Config config) throws IOException {
```

Comment:
The code below seems to assume that all of the configuration is mandatory. 
Should we add defaults to most of the config values?